lib/client/BootstrappedDataService.js

```diff
@@ -58,6 +53,14 @@ BootstrappedDataService.prototype = {
 
 };
 
+BootstrappedDataService.computeKey = function(url, queryParams) {
+    if (queryParams) {
+        return url + JSON.stringify(queryParams);
```

Comment:
So I looked into creating a query string. It is trivial on the server since Node.js has built-in query string support, however, we need to resolve objects to the same string on the client at look-up time.

For the server side performance, I created this test:
```javascript
"use strict";

var querystring = require("querystring");

var TEST_OBJ = {
    a: 1,
    b: {
        c: [1, 2],
        d: "a string"
    }
}

function test1() {
    return JSON.stringify(TEST_OBJ);
}

function test2() {
    return querystring.stringify(TEST_OBJ);
}

var tests = [
    test1,
    test2
];

var SAMPLE_SIZE = 1000000;

tests.forEach(function(testFn, index) {
    console.time(index);

    for (var i = 0; i < SAMPLE_SIZE; ++i) {
        testFn();
    }

    console.timeEnd(index);
});
```

`querystring.stringify` was about 30% slower on the server than `JSON.stringify`.